src/tree/mod.rs

```diff
@@ -286,26 +341,66 @@ where
         // Get the latest root from all bridged chains
         let latest_bridged_roots = self.latest_bridged_roots().await?;
 
-        // Update chain state for bridged roots
-        for root in identity_updates.keys() {
-            if let Some(chain_ids) = latest_bridged_roots.get(&root.hash) {
+        if identity_updates.is_empty() {
+            // @dev If there are no identity updates, meaning that the in-memory tree's latest root matches the onchain root across all chains.
+            // In this case, we can set all chains to the latest root, with the root nonce set to 0. When the next canonical update is received,
+            // the root for the canonical chain_id will be updated and once the new root is bridged to all chains,
+            // the pending tree_updates will be applied and the root with nonce 0 will no longer be in the chain state hashmap.
+            let latest_root = identity_tree.tree.root();
+
+            let root = Root {
+                hash: latest_root,
+                nonce: 0,
+            };
+
+            // If the latest bridged roots is empty, this means that we are not monitoring any bridged chains
+            // and we do not need to check the following
+            if !latest_bridged_roots.is_empty() {
+                let chain_ids = latest_bridged_roots
+                    .get(&latest_root)
+                    .expect("No bridged roots match the latest canonical root");
+
+                // Ensure that all bridged chains have the same root
+                if chain_ids.len() != self.bridged_tree_manager.len() {
+                    return Err(eyre::eyre!(
```

Comment:
Not sure if you've seen the `bail!` macro, but it's pretty handy!

Reply:
Yeah,`bail!` is nicer here but this will be removed once the error handling PR is merged anyway.